app/javascript/mastodon/components/hotkeys.jsx

```diff
@@ -0,0 +1,119 @@
+  useRegisterHotkey(Array.isArray(keyMap.toggleComposeSpoilers) ? keyMap.toggleComposeSpoilers : [keyMap.toggleComposeSpoilers], 'toggleComposeSpoilers');
+  useRegisterHotkey(Array.isArray(keyMap.focusColumn) ? keyMap.focusColumn : [keyMap.focusColumn], 'focusColumn');
+  useRegisterHotkey(Array.isArray(keyMap.reply) ? keyMap.reply : [keyMap.reply], 'reply');
+  useRegisterHotkey(Array.isArray(keyMap.favourite) ? keyMap.favourite : [keyMap.favourite], 'favourite');
+  useRegisterHotkey(Array.isArray(keyMap.boost) ? keyMap.boost : [keyMap.boost], 'boost');
+  useRegisterHotkey(Array.isArray(keyMap.mention) ? keyMap.mention : [keyMap.mention], 'mention');
+  useRegisterHotkey(Array.isArray(keyMap.open) ? keyMap.open : [keyMap.open], 'open');
+  useRegisterHotkey(Array.isArray(keyMap.openProfile) ? keyMap.openProfile : [keyMap.openProfile], 'openProfile');
+  useRegisterHotkey(Array.isArray(keyMap.moveDown) ? keyMap.moveDown : [keyMap.moveDown], 'moveDown');
+  useRegisterHotkey(Array.isArray(keyMap.moveUp) ? keyMap.moveUp : [keyMap.moveUp], 'moveUp');
+  useRegisterHotkey(Array.isArray(keyMap.back) ? keyMap.back : [keyMap.back], 'back');
+  useRegisterHotkey(Array.isArray(keyMap.goToHome) ? keyMap.goToHome : [keyMap.goToHome], 'goToHome');
+  useRegisterHotkey(Array.isArray(keyMap.goToNotifications) ? keyMap.goToNotifications : [keyMap.goToNotifications], 'goToNotifications');
+  useRegisterHotkey(Array.isArray(keyMap.goToLocal) ? keyMap.goToLocal : [keyMap.goToLocal], 'goToLocal');
+  useRegisterHotkey(Array.isArray(keyMap.goToFederated) ? keyMap.goToFederated : [keyMap.goToFederated], 'goToFederated');
+  useRegisterHotkey(Array.isArray(keyMap.goToDirect) ? keyMap.goToDirect : [keyMap.goToDirect], 'goToDirect');
+  useRegisterHotkey(Array.isArray(keyMap.goToStart) ? keyMap.goToStart : [keyMap.goToStart], 'goToStart');
+  useRegisterHotkey(Array.isArray(keyMap.goToFavourites) ? keyMap.goToFavourites : [keyMap.goToFavourites], 'goToFavourites');
+  useRegisterHotkey(Array.isArray(keyMap.goToPinned) ? keyMap.goToPinned : [keyMap.goToPinned], 'goToPinned');
+  useRegisterHotkey(Array.isArray(keyMap.goToProfile) ? keyMap.goToProfile : [keyMap.goToProfile], 'goToProfile');
+  useRegisterHotkey(Array.isArray(keyMap.goToBlocked) ? keyMap.goToBlocked : [keyMap.goToBlocked], 'goToBlocked');
+  useRegisterHotkey(Array.isArray(keyMap.goToMuted) ? keyMap.goToMuted : [keyMap.goToMuted], 'goToMuted');
+  useRegisterHotkey(Array.isArray(keyMap.goToRequests) ? keyMap.goToRequests : [keyMap.goToRequests], 'goToRequests');
+  useRegisterHotkey(Array.isArray(keyMap.toggleHidden) ? keyMap.toggleHidden : [keyMap.toggleHidden], 'toggleHidden');
+  useRegisterHotkey(Array.isArray(keyMap.toggleSensitive) ? keyMap.toggleSensitive : [keyMap.toggleSensitive], 'toggleSensitive');
+  useRegisterHotkey(Array.isArray(keyMap.openMedia) ? keyMap.openMedia : [keyMap.openMedia], 'openMedia');
```

Comment:
This seems pretty verbose.

Also, I am not sure everything gets properly memoized and there is no duplicate work being performed at every render. But I've only glanced at it and I'm not familiar with `react-hotkeys-hook`, so maybe those remarks are unwarranted.